core-web/libs/portlets/dot-ema/src/lib/edit-ema-editor/components/edit-ema-palette/store/edit-ema-palette.store.ts

```diff
@@ -0,0 +1,201 @@
+
+            return data$.pipe(
+                switchMap(({ allowedContent, filter }) => {
+                    return this.getFilteredContentTypes(filter, allowedContent).pipe(
+                        tapResponse(
+                            (contentTypes) => {
+                                this.patchState({
+                                    contenttypes: { items: contentTypes, filter },
+                                    status: EditEmaPaletteStoreStatus.LOADED
+                                });
+                            },
+                            (err) => {
+                                this.setStatus(EditEmaPaletteStoreStatus.ERROR);
+                                // eslint-disable-next-line no-console
+                                console.log(err);
+                            }
+                        )
+                    );
+                })
+            );
+        }
+    );
+
+    /**
+     * Loads contentlets based on the provided filter, content type name, language ID, and page number.
+     * @param data$ An observable that emits an object containing the filter, content type name, language ID, and page number.
+     * @returns An observable that emits the loaded contentlets.
+     */
+    readonly loadContentlets = this.effect(
+        (
+            data$: Observable<{
+                filter: string;
+                contenttypeName: string;
+                languageId: string;
+                page?: number;
+            }>
+        ) => {
+            return data$.pipe(
+                tap(() => this.setStatus(EditEmaPaletteStoreStatus.LOADING)),
+                switchMap(({ filter, contenttypeName, languageId, page }) => {
+                    return this.dotESContentService
+                        .get({
+                            itemsPerPage: PALETTE_PAGINATOR_ITEMS_PER_PAGE,
+                            lang: languageId || '1',
+                            filter: filter || '',
+                            offset:
+                                ((page || 0) * PALETTE_PAGINATOR_ITEMS_PER_PAGE).toString() || '0',
+                            query: `+contentType: ${contenttypeName} +deleted: false`.trim()
+                        })
+                        .pipe(
+                            tapResponse(
+                                (contentlets) => {
+                                    this.patchState({
+                                        contentlets: {
+                                            items: contentlets.jsonObjectView.contentlets,
+                                            filter: {
+                                                query: filter,
+                                                contentTypeVarName: contenttypeName
+                                            },
+                                            totalRecords: contentlets.resultsSize,
+                                            itemsPerPage: PALETTE_PAGINATOR_ITEMS_PER_PAGE
```

Comment:
Do we need to set this everytime?

Reply:
As a nested state, yes, the patchState on nested needs to have all fields